File: back/src/com/servlet/SignUp.java
package com.servlet;

import com.dao.UserDao;
import com.entity.User;

import javax.servlet.annotation.*;
import javax.servlet.http.*;
import javax.servlet.*;
import java.io.IOException;

@WebServlet("/SignUp")
public class SignUp extends HttpServlet {
    @Override
    protected void doGet(HttpServletRequest httpServletRequest,HttpServletResponse httpServletResponse) throws IOException,ServletException
    {
        this.doPost(httpServletRequest,httpServletResponse);
    }

    @Override
    protected void doPost(HttpServletRequest httpServletRequest,HttpServletResponse httpServletResponse) throws IOException,ServletException
    {
        httpServletRequest.setCharacterEncoding("utf-8");
        httpServletResponse.setCharacterEncoding("utf-8");//设定编码防止中文乱码
        httpServletResponse.setContentType("text/plain;charset=utf-8");//设置相应类型为html,编码为utf-8

        String name = httpServletRequest.getParameter("name");//根据name获取参数
        String password = httpServletRequest.getParameter("password");//根据password获取参数

        UserDao userDao = new UserDao();
        User user = new User();
        user.setName(name);
        user.setPassword(password);

        if(!userDao.add(user)) //若添加失败
        {
            httpServletResponse.sendError(204,"add failed.");//设置204错误码与出错信息
        }
    }
}

File: KotlinDemo/JavaWebDemo/src/main/kotlin/com/example/javawebdemo/servlet/SignUp.kt
package com.example.javawebdemo.servlet

import com.example.javawebdemo.dao.Dao
import com.example.javawebdemo.entity.User
import com.example.javawebdemo.response.ResponseBody
import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.annotation.WebServlet
import jakarta.servlet.http.HttpServlet
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

@WebServlet("/sign/up")
class SignUp : HttpServlet() {
    override fun doPost(req: HttpServletRequest?, resp: HttpServletResponse?) {
        println("doPost in sign up")
        println(req == null)
        println(resp == null)
        if (req != null && resp != null) {
            req.characterEncoding = "utf-8"
            resp.characterEncoding = "utf-8"
            resp.contentType = "application/json;charset=utf-8"
            val name = req.getParameter("name")
            val password = req.getParameter("password")

            println(name)
            println(password)

            val dao = Dao()
            val user = User(name, password)
            val mapper = ObjectMapper()
            val body = ResponseBody()

            if (dao.insert(user)) {
                body.code = 200
                body.data = "success"
            } else {
                body.code = 500
                body.data = "failed"
            }
            mapper.writeValue(resp.writer, body)
        }
    }
}